File: pms-backend/src/main/java/com/pms/service/AppointmentService.java
package com.pms.service;

import com.pms.dto.AppointmentDto;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.List;

public interface AppointmentService {
    AppointmentDto createAppointment(AppointmentDto dto);
    AppointmentDto updateAppointment(Long id, AppointmentDto dto);
    AppointmentDto updateAppointmentStatus(Long id, String status);
    AppointmentDto getAppointmentById(Long id);
    List<AppointmentDto> getAppointmentsByPatientId(Long patientId);
    List<AppointmentDto> getAppointmentsByDoctorId(Long doctorId);
    Page<AppointmentDto> searchAppointments(Long patientId, Long doctorId, Long departmentId, String status, LocalDate fromDate, LocalDate toDate, String query, Pageable pageable);
    void cancelAppointment(Long id);
    List<LocalTime> getAvailableDoctorSlots(Long doctorId, LocalDate date);
}

File: pms-frontend/src/services/appointmentService.js
import api from './api';

export const appointmentService = {
  searchAppointments: async (params = {}) => {
    const res = await api.get('/appointments', { params });
    return res.data;
  },

  getAppointmentById: async (id) => {
    const res = await api.get(`/appointments/${id}`);
    return res.data;
  },

  getAppointmentsByPatientId: async (patientId) => {
    const res = await api.get(`/appointments/patient/${patientId}`);
    return res.data;
  },

  getAppointmentsByDoctorId: async (doctorId) => {
    const res = await api.get(`/appointments/doctor/${doctorId}`);
    return res.data;
  },

  getAvailableSlots: async (doctorId, date) => {
    const res = await api.get('/appointments/available-slots', { params: { doctorId, date } });
    return res.data;
  },

  createAppointment: async (data) => {
    const res = await api.post('/appointments', data);
    return res.data;
  },

  updateAppointment: async (id, data) => {
    const res = await api.put(`/appointments/${id}`, data);
    return res.data;
  },

  updateAppointmentStatus: async (id, status) => {
    const res = await api.patch(`/appointments/${id}/status`, { status });
    return res.data;
  },

  cancelAppointment: async (id) => {
    const res = await api.delete(`/appointments/${id}`);
    return res.data;
  }
};
